src/src/EmailBundle/Controller/EmailController.php

```diff
@@ -26,27 +26,32 @@ public function sendEmail($attendee, array $context, String $templateName, Strin
         $plain = $this->renderToString($templateName, 'emails/'.$emailType.'.txt.twig', $context);
         $html = $this->renderToString($templateName, 'emails/'.$emailType.'.html.twig', $context);
         $meta = $this->getEmailMeta($templateName, 'emails/'.$emailType.'.yaml.twig', $context);
-
-        $message = (new \Swift_Message($meta['subject']))
-            ->setFrom($meta['email_from'])
-            ->setTo($attendee->getEmail())
-            ->setBody($html, 'text/html')
-            ->addPart($plain, 'text/plain')
-        ;
-
-        $status = $this->get('mailer')->send($message);
-
-        $log = new EmailLog();
-        $log->setTemplate($templateName);
-        $log->setContentHtml($html);
-        $log->setContentPlain($plain);
-        $log->setEmailAddress($attendee->getEmail());
-        $log->setEmailMeta(json_encode($meta));
-        $log->setEmailType($emailType);
-        $log->setStatus($status);
+        $email = $attendee->getEmail();
+        $encodedMeta = json_encode($meta);
 
         $em = $this->getDoctrine()->getManager();
-        $em->persist($log);
-        $em->flush();
+        $isSent = $em->getRepository(EmailLog::class)->isEmailSent($templateName, $html, $plain, $email, $encodedMeta, $emailType);
+        if (!$isSent) {
```

Comment:
dal by som skor `if($isSent) {return;}` aby to nebolo cele take vnorene, ale je to len moja osobna preferencia.

Reply:
Ak sa ti to tak paci, tak to tam kludne dam